lms/static/scripts/frontend_apps/components/test/AutoGradingConfigurator-test.js

```diff
@@ -0,0 +1,172 @@
+          fakeUpdateAutoGradingConfig,
+          sinon.match({ activityCalculation }),
+        );
+      });
+
+      it('renders inputs based on activity calculation value', () => {
+        fakeAutoGradingConfig.activityCalculation = activityCalculation;
+
+        const wrapper = createComponent();
+        const inputs = wrapper.find('AnnotationsGoalInputGroup');
+        const firstInput = inputs.first();
+
+        assert.equal(
+          activityCalculation === 'separately' ? 2 : 1,
+          inputs.length,
+        );
+        assert.equal(
+          `Annotations${activityCalculation === 'cumulative' ? ' + replies' : ''}Minimum`,
+          firstInput.text(),
+        );
+      });
+    });
+
+    ['all_or_nothing', 'scaled'].forEach(gradingType => {
+      it('updates config when changing grading type', () => {
+        const wrapper = createComponent();
+
+        dispatchOnChange(
+          wrapper,
+          '[data-testid="grading-type-radio-group"]',
+          gradingType,
+        );
+
+        assert.calledWith(
+          fakeUpdateAutoGradingConfig,
+          sinon.match({ gradingType }),
+        );
+      });
+
+      it('renders different input label depending on grading type value', () => {
+        fakeAutoGradingConfig.gradingType = gradingType;
+
+        const wrapper = createComponent();
+        const input = wrapper.find('AnnotationsGoalInputGroup').first();
+
+        assert.isTrue(
+          input
+            .text()
+            .endsWith(gradingType === 'all_or_nothing' ? 'Minimum' : 'Goal'),
+        );
+      });
+    });
+
+    [
+      {
+        inputIndex: 0,
+        value: '15',
+        expectedConfig: { requiredAnnotations: 15 },
+      },
+      {
+        inputIndex: 1,
+        value: '3',
+        expectedConfig: { requiredReplies: 3 },
+      },
+    ].forEach(({ inputIndex, value, expectedConfig }) => {
+      it('updates config when inputs change', () => {
+        fakeAutoGradingConfig.activityCalculation = 'separately';
+
+        const wrapper = createComponent();
+        const inputs = wrapper.find('AnnotationsGoalInputGroup');
+
+        act(() =>
+          inputs.at(inputIndex).find('Input').props().onChange({
+            target: { value },
+          }),
+        );
+
+        assert.calledWith(
+          fakeUpdateAutoGradingConfig,
+          sinon.match(expectedConfig),
+        );
+      });
+    });
+  });
+
+  it(
+    'should pass a11y checks',
+    checkAccessibility([
+      {
+        name: 'inactive',
+        content: () => createComponent(),
+      },
+      // FIXME We have an unrelated accessibility issue on RadioGroup, where
+      //       selected Radio's subtitles do not have enough contrast with the
+      //       background color. I'll fix that separately
+      // {
+      //   name: 'active',
+      //   content: () => {
+      //     fakeAutoGradingConfig.active = true;
+      //     return createComponent();
+      //   },
+      // },
```

Comment:
I commented this out for now. I'm trying to fix the contrast accessibility issue in parallel.

See the conversation in slack for context https://hypothes-is.slack.com/archives/C1M8NH76X/p1725351905374359